app/src/ui/branches/pull-request-list.tsx

```diff
@@ -48,19 +54,35 @@ export class PullRequestList extends React.Component<
   public constructor(props: IPullRequestListProps) {
     super(props)
 
+    const group = createListItems(props.pullRequests)
+    const pullRequest = props.currentPullRequest
+    const selectedItem = pullRequest
+      ? findItemForPullRequest(group, pullRequest)
+      : null
+
     this.state = {
-      groupedItems: [createListItems(props.pullRequests)],
+      groupedItems: [group],
       filterText: '',
-      selectedItem: null,
+      selectedItem: selectedItem || null,
     }
   }
 
   public componentWillReceiveProps(nextProps: IPullRequestListProps) {
-    if (nextProps.pullRequests === this.props.pullRequests) {
+    if (
+      nextProps.pullRequests === this.props.pullRequests &&
+      nextProps.currentPullRequest === this.props.currentPullRequest
+    ) {
       return
     }
 
-    this.setState({ groupedItems: [createListItems(nextProps.pullRequests)] })
+    const group = createListItems(nextProps.pullRequests)
+    const pullRequest = nextProps.currentPullRequest
+    const selectedItem = pullRequest
+      ? findItemForPullRequest(group, pullRequest)
```

Comment:
Sorry if I'm being picky here but won't this potentially steal focus when background updates happen? Thinking about the scenario where I'm using my keyboard in the PR list to move around and a background update happens which creates new pull request objects but the current pull request id hasn't changed. Should we perhaps introduce an equality function here which compares on `.id` instead?